test/eventsource_test.js

```diff
@@ -466,6 +466,24 @@ describe('Parser', function () {
       }
     })
   })
+
+  it('parses a relatively huge message across many chunks efficiently', function (done) {
+    this.timeout(1000)
+
+    createServer(function (err, server) {
+      if (err) return done(err)
+
+      var longMessage = 'data: ' + new Array(100000).join('a') + '\n\n'
+      var longMessageChunks = longMessage.match(/[\s\S]{1,10}/g) // Split the message into chunks of 10 characters
+      server.on('request', writeEvents(longMessageChunks))
+
+      var es = new EventSource(server.url)
+
+      es.onmessage = function () {
+        server.close(done)
```

Comment:
I'd feel a lot more comfortable if the test made assertions about the received messages. Could you add that please?

Reply:
Sure thing. I added an assertion to verify the message contents.

Reply:
This seems fine to me, @aslakhellesoy - any objections on merging?

Reply:
Looks good to me. @rexxars can you please update `HIstory.md` whenever a change/improvement lands in master? Thanks!

Reply:
Will do, @aslakhellesoy